src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -632,12 +633,21 @@ ReactDOMComponent.Mixin = {
           this._currentElement.type
         );
       }
+      var isCustomComponentTag = isCustomComponent(this._tag, props);
+      if (__DEV__ && isCustomComponentTag && !didWarnShadyDOM && el.shadyRoot) {
+        warning(
+          false,
+          'A component is using shady DOM. Using shady DOM with React can ' +
```

Comment:
Can we put the component name here please?
You should be able to do something like:

```js
var type = this._currentElement.type;
var name = type.displayName || type.name || 'A component';
warning(
  false,
  '%s is using shady DOM. (...)',
  name
);
```